src/script/api/script_vehicle.hpp

```diff
@@ -97,9 +97,19 @@ class ScriptVehicle : public ScriptObject {
 	 * Checks whether the given vehicle is valid and owned by you.
 	 * @param vehicle_id The vehicle to check.
 	 * @return True if and only if the vehicle is valid.
+	 * @note Also returns true when the leading part of the vehicle is a wagon.
```

Comment:
It might be useful to direct people to use IsPrimaryVehicle() instead.
```suggestion
	 * @note Also returns true when the leading part of the vehicle is a wagon. Use IsPrimaryVehicle() to check for a valid vehicle with a leading engine.
```